components/kubeflow-pipeline/python/fl_suite/analytics/_histogram.py

```diff
@@ -0,0 +1,154 @@
+    cleanup_kubernetes_resources,
+    create_image_tag,
+    setup_kubernetes_resources,
+)
+
+
+# pylint: disable-next=too-many-arguments
+def histogram(
+    data_func: Callable[[str, bool], ContainerOp],
+    nbins: int,
+    hrange: Tuple[float, float],
+    min_available_clients: int = 2,
+    host: Optional[str] = None,
+    experiment_name: Optional[str] = None,
+    registry: str = "ghcr.io/katulu-io/fl-suite",
+    verify_registry_tls: bool = True,
+) -> None:
+    """Run distributed histogram of data provided by multiple clients."""
+    analytics_server = f"{registry}/analytics-server:{__version__}"
+
+    client = Client(host)
+    pipeline_run = client.create_run_from_pipeline_func(
+        histogram_pipeline(
+            fl_client=data_func,
+            fl_server_image=analytics_server,
+            registry=registry,
+            verify_registry_tls=verify_registry_tls,
+        ),
+        arguments={
+            "min_available_clients": min_available_clients,
+            "nbins": nbins,
+            "hmin": hrange[0],
+            "hmax": hrange[1],
+        },
+        experiment_name=experiment_name,
+    )
+    result = client.wait_for_run_completion(pipeline_run.run_id, timeout=3600)
+    status = result.run.status.lower()
+
+    if status in ["failed", "skipped", "error"]:
+        raise RuntimeError(f"Run {status}")
+
+
+def histogram_server(
+    server_image: str,
+    min_available_clients: int,
+    nbins: int,
+    hrange: Tuple[float, float],
+) -> ContainerOp:
+    """Component to run a Flower server for federated histogram."""
+    analytics_server_spec = ComponentSpec(
+        name="Flower Server",
+        description="Start a Flower server for Federated Analytics",
+        metadata=MetadataSpec(
+            labels={
+                "katulu/fl-server": "flower-server",
+            },
+        ),
+        inputs=[
+            InputSpec("min_available_clients", type="Integer"),
+            InputSpec("nbins", type="Integer"),
+            InputSpec("hmin", type="Float"),
+            InputSpec("hmax", type="Float"),
+        ],
+        outputs=[
+            OutputSpec("MLPipeline_ui_metadata", type="UI metadata"),
+        ],
+        implementation=ContainerImplementation(
+            ContainerSpec(
+                image=server_image,
+                command=[
+                    "python3",
+                    "/pipelines/component/flwr_analytics_server",
+                    "--min-available-clients",
+                    InputValuePlaceholder("min_available_clients"),
+                    "--metadata-output-path",
+                    OutputPathPlaceholder("MLPipeline_ui_metadata"),
+                    "histogram",
+                    "--nbins",
+                    InputValuePlaceholder("nbins"),
+                    "--hmin",
+                    InputValuePlaceholder("hmin"),
+                    "--hmax",
+                    InputValuePlaceholder("hmax"),
+                ],
```

Comment:
You are already setting an entrypoint (`command` in `kfp` lingo) for the `flwr_analytics_server` Dockerfile. You could then assign just the args you want to use:

```suggestion
                args=[
                    "--min-available-clients",
                    InputValuePlaceholder("min_available_clients"),
                    "--metadata-output-path",
                    OutputPathPlaceholder("MLPipeline_ui_metadata"),
                    "histogram",
                    "--nbins",
                    InputValuePlaceholder("nbins"),
                    "--hmin",
                    InputValuePlaceholder("hmin"),
                    "--hmax",
                    InputValuePlaceholder("hmax"),
                ],
```

Reply:
Based on the above comment, we could then define a a common `analytics_server` function that accepts different arguments we want to pass, and assign the specific histogram and correlation args. For now this is not a show stopper from my side and we can implement that in a following PR.

Reply:
Unfortunately, we need to have the full command here again, as `kfp` is overwriting whatever is the entrypoint in `Dockerfile`. That is because it's wrapping the actual command in its own command to be able to inject `kfp` specific stuff.

Reply:
You are right!. I remember that we had a similar issue with the envoyproxy sidecars, hence the need to repeat the command ([source](https://github.com/katulu-io/fl-suite/blob/beta/components/kubeflow-pipeline/python/fl_suite/pipelines/_flower_server.py#L98); also mentioned in kubeflow's [docs](https://www.kubeflow.org/docs/components/pipelines/installation/choose-executor/#emissary-executor)). Nevertheless the refactoring I mentioned about a new function called `analytics_server` that can be used for the `histogram_server` and `correlation_server` still applies.